starknet_py/net/l1/messages.py

```diff
@@ -0,0 +1,169 @@
+from dataclasses import dataclass
+from functools import reduce
+from typing import List, Optional
+from eth_utils import keccak
+from hexbytes import HexBytes
+from starkware.starknet.public.abi import get_selector_from_name
+
+from starknet_py.net.l1.contracts import StarknetL1Contract
+from starknet_py.net.models import (
+    StarknetChainId,
+    EthBlockIdentifier,
+    AddressRepresentation,
+    parse_address,
+)
+from starknet_py.utils.sync import add_sync_methods
+
+
+def encode_packed(*args: List[int]) -> bytes:
+    return reduce(
+        lambda acc, x: acc + x,
+        [x.to_bytes(32, byteorder="big", signed=False) for x in args],
+    )
+
+
+@dataclass
+class L1MessageContent:
+    """A dataclass representing a message's payload (from L1 to L2)"""
+
+    l2_sender: AddressRepresentation
+    l1_recipient: int  # Integer representation of l1 hex address
+    payload: List[int]
+
+    @property
+    def hash(self):
+        return keccak(
+            encode_packed(
+                parse_address(self.l2_sender),
+                self.l1_recipient,
+                len(self.payload),
+                *self.payload,
+            )
+        )
+
+
+def int_from_hexbytes(hexb: HexBytes) -> int:
+    return int(hexb.hex(), 16)
+
+
+@add_sync_methods
+@dataclass
+class L1Message:
+    """
+
+    Class which enables you to check L2 -> L1 messages status
+    """
+
+    hash: bytes  # 32 bytes hash representation
+
+    @classmethod
+    def from_hash(cls, msg_hash: bytes) -> "L1Message":
+        """
+
+        :param msg_hash: Message's hash
+        :return: Instance of `L1Message`
+        """
+        return cls(hash=msg_hash)
+
+    @classmethod
+    def from_content(cls, msg_content: L1MessageContent) -> "L1Message":
```

Comment:
I think that for this to be useful, there should also be a way to extract `L1MessageContent` from the transaction receipt.
Otherwise it is very hard for me as a user to know the exact L1MessageContent that I've sent.